### PR TITLE
Version Generation refactor

### PR DESCRIPTION
**Pull request by callicles**
This refactors the version generation to automatically increment the patch version unless something else is specified as well as generate changelogs from PRs.

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -31,3 +31,20 @@ $ pnpm install
...
+
+We use [semantic versioning](https://semver.org/) to denote versions of the different components of the system.
+
+We have automation that helps keep a really high cadence as we develop the initial version of the framework and ecosystem. 
+
+> We might change how we release later in the lifecycle of the project.
+
+Currently we release every time code that changes the framework or the CLI is merged to the `main` branch. We are [releasing from the trunk](https://trunkbaseddevelopment.com/release-from-trunk/).
+
+* By Default, every commit on `main` will increase the patch version. ie in `x.y.(z+1)`
```
Comment (tg339) [inline CONTRIBUTING:45]: 🙌

File: apps/igloo-cli-npm/README.md [hunk 2]
```diff
@@ -12,16 +15,12 @@ The Igloo CLI is available as an NPM or Pip package. To install the CLI, run the
 npm install -g @igloo-kit/cli
 ```
 
-or 
-
-```bash
-pip install igloo-cli
-```
```
Comment (tg339) [inline apps/igloo-cli-npm/README.md:19]: Why did you remove python?

Comment (callicles) [reply apps/igloo-cli-npm/README.md:19]: Because this is the readme only for NPM - we can add to the main readme if you want

Comment (callicles) [reply apps/igloo-cli-npm/README.md:19]: Added it to the main readme
